### PR TITLE
feat: add util.cid options

### PR DESCRIPTION
**Pull request by richardschneider**
See https://github.com/ipld/interface-ipld-format/issues/40

Closes #62

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -108,10 +108,32 @@ exports.deserialize = (data, callback) => {
...
+ *
+ * @param {Object} dagNode - Internal representation
+ * @param {Object} [options] - Options to create the CID
+ * @param {number} [options.version=1] - CID version number
+ * @param {string} [options.hashAlg] - Defaults to hashAlg for the resolver
+ * @param {CidCallback} callback - Callback that handles the return value
+ * @returns {void}
+ */
+exports.cid = (dagNode, options, callback) => {
+  if (options instanceof Function) {
```
Comment (vmx) [inline src/util.js]: Searching through the existing IPFS/IPLD code base it seems that the preferred way for checking if something is a function is: ` if (typeof options !== 'function') {`. My guess is that it is related to code minifiers. Please change it.

File: src/util.js [hunk 2]
```diff
@@ -108,10 +108,32 @@ exports.deserialize = (data, callback) => {
...
+    callback = options
+    options = {}
+  }
+  options = options || {}
+  const hashAlg = options.hashAlg || resolver.defaultHashAlg
+  const version = options.version || 1
```
Comment (vmx) [inline src/util.js]: This doesn't work correctly. If `options.version` is set to `0`, then `version` would become `1` as `0` evaluates to false.

**Comment by richardschneider** [comment]
The only error cases, that I can think of,  are an invalid version or hash algorithm.  Both cases should be handled by by underlaying software, `cids` and `multihashing-async`.

**Comment by vmx** [comment]
Please squash merge (as mentioned on the other PRs :)

**Comment by richardschneider** [comment]
I'll look into the libraries and report back. I'm `makaretu` on IRC, what is your nickname,

**Comment by vmx** [comment]
I'm `vmx` on IRC.

**Pull request merged**